### PR TITLE
Create registration page

### PR DESCRIPTION
**Pull request by Kalatco**

### CODE REVIEW

**Comment by Kalatco** [comment]
@kansairob please make the font size larger, its very small on the phone compared to the size of the page

**Comment by Kalatco** [comment]
@kansairob on the login view, when the username or password is incorrect, a message is shown to the user, the registration form needs it too.
`<View style={{ flex: 1, position: "relative" }}>
        {isInvalidLogin && (
            <View>
              <Text style={styles.invalidCredentials}>
                User email or password is incorrect
              </Text>
            </View>
          )}
      </View>`
Please look at how the login view is handling this and make a new one for the registration view.

**Comment by Kalatco** [comment]
> @kansairob please make the font size larger, its very small on the phone compared to the size of the page

File: src/views/LoginView.js [hunk 1]
```diff
@@ -60,14 +70,64 @@ function SigninView(props) {
...
+    formData.append('password2', confirm_passwordReg);
+    formData.append('first_name', first_nameReg);
+    formData.append('last_name', last_nameReg);
+
+    axios.post(`${props.serverName}/register`, formData)
+      .then((res) => {
+        props.setAuthToken(res.data.token);
+
+        // Once the auth token is received, access user data
+        if (res.data.token) {
```
Comment (Kalatco) [inline src/views/LoginView.js:87]: Found issue with bad registration route, it throws a "200" error (which is technically an accept code, no idea why this is happening), but if the request doesn't have `res.data.token`, create an else case that shows the specific message using:
```
       } else {
          // Error occured in 200 response, display invalid fields to user
          setIsInvalidRegistration(true);
          let errorMessage = "Invalid: \n";
          for (const [key, value] of Object.entries(res.data)) {

            if (Array.isArray(value)) {
              errorMessage += `${key.toString()}: ${value[0]}\n`
            } else {
              errorMessage += `${key.toString()}\n`
            }
          }
          setInvalidRegistrationMessage(errorMessage);
        }
```

and have these use states:
```
const [isInvalidRegistration, setIsInvalidRegistration] = useState(false);
  const [InvalidRegistrationMessage, setInvalidRegistrationMessage] = useState("");
```

and for the error message:

```
{isInvalidRegistration && (
            <View>
              <Text style={styles.invalidCredentials}>
                { InvalidRegistrationMessage }
              </Text>
            </View>
          )}
```

But if the server does throw a 400 or 500 error, in the catch statement for the registration API request, add:
```
.catch((res) => {
        setIsInvalidRegistration(true);
        setInvalidRegistrationMessage("Server issue encountered");
      })
```

File: src/views/LoginView.js [hunk 2]
```diff
@@ -136,20 +281,45 @@ const styles = StyleSheet.create({
...
+    marginHorizontal: 5,
     flex: 1,
   },
   inputStyle: {
-    
+    color: "black",
+    fontWeight: "bold", 
+    justifyContent: "center",
+    textAlign: "left",
+    fontSize: 18,
```
Comment (Kalatco) [inline src/views/LoginView.js:294]: the font size is good for the title of each input field, but not so good for the text inputs, the title and input fields are blending in with each other, make a new style for the text inputs and make them smaller

File: src/views/LoginView.js [hunk 3]
```diff
@@ -95,7 +169,78 @@ function SigninView(props) {
...
+            onChangeText={(input) => setEmailReg(input)}/> 
+
+          <Text style={styles.inputStyle}>Username:</Text>
+          <TextInput  
+            placeholder="username"
+            style={styles.inputStyle}
+            autoCorrect={false}
+            onChangeText={(input) => setUsernameReg(input)}/>  
+
+          <Text style={styles.inputStyle}>Password:</Text>
```
Comment (Kalatco) [inline src/views/LoginView.js:188]: both password fields are showing the user's password, these need to be hidden

Comment (Kalatco) [reply src/views/LoginView.js:188]: @kansairob

File: src/views/LoginView.js (hunk 2 above)
Comment (Kalatco) [reply src/views/LoginView.js:294]: @kansairob

File: src/views/LoginView.js (hunk 1 above)
Comment (Kalatco) [reply src/views/LoginView.js:87]: Capture
This is what the output will look like, BUT YOU NEED TO MAKE SURE IT RUNS ON YOUR DEVICE BEFORE PUSHING. do not worry about the offset buttons, this happens when the error message has a lot of text. there is not enough time to worry about this @kansairob

Comment (kansairob) [reply src/views/LoginView.js:87]: sure no problem

Comment (kansairob) [reply src/views/LoginView.js:88]: All changes have been updated in the file.

File: src/views/LoginView.js (hunk 2 above)
Comment (kansairob) [reply src/views/LoginView.js:311]: New style was created for the text input called textInput

File: src/views/LoginView.js (hunk 3 above)
Comment (kansairob) [reply src/views/LoginView.js:203]: Done

**Pull request merged**